presto-hive-metastore/src/main/java/com/facebook/presto/hive/metastore/thrift/HiveMetastore.java

```diff
@@ -72,6 +73,11 @@
 
     List<String> getPartitionNamesByFilter(String databaseName, String tableName, Map<Column, Domain> partitionPredicates);
 
+    default List<PartitionNameWithVersion> getPartitionNamesWithVersionByFilter(String databaseName, String tableName, Map<Column, Domain> partitionPredicates)
```

Comment:
I'm a bit hesitated about this approach because this method is not compatible with open source hive metastore, but adding as a default method might be okay.

One other option is to add cache functionality in our internal Hive metastore extension. The pro is that everything is hidden under the hood so we likely need zero changes in open source, but the con is that there would be a bit of logic duplication around caching, and we won't be able to pass version information around if needed.